database/migrations/001-punishments-schema.sql

```diff
@@ -0,0 +1,46 @@
+--------------------------------------------------------------------------------
+-- Up
+--------------------------------------------------------------------------------
+
+CREATE TABLE IF NOT EXISTS punishments (
+	punishType TEXT NOT NULL,
+	userid TEXT NOT NULL,
+	ips TEXT NOT NULL,
+	userids TEXT NOT NULL,
```

Comment:
Are you using this field as an text "array" (e.g., multiple elements separated with a delimiter)? The plural gives that impression. Since this is not the way to do arrays in SQL databases.

You're meant to represent arrays by having each element be a separate row, with elements belonging to the same array sharing a key. Aka a [join table](https://en.wikipedia.org/wiki/Associative_entity). See [this](https://stackoverflow.com/questions/17371639/how-to-store-arrays-in-mysql) for a concrete example

Reply:
Hm, I'm not sure if it's worth having extra tables, adding data to two extra tables, etc.

Are there any benefits with going full relational database rather than just serializing / deserializing?

Reply:
Does SQLite just have TEXT and VARCHAR and all those just work exactly the same?

Reply:
I wonder if this should be a JSON column. Apparently SQLite supports those.

Reply:
> Does SQLite just have TEXT and VARCHAR and all those just work exactly the same?

https://www.sqlite.org/datatype3.html

Reply:
The benefits of going full relational is being able to query individual elements. E.g., Look up which punishments apply to a given IP